common/src/main/java/ac/grim/grimac/utils/data/TeleportData.java

```diff
@@ -12,9 +12,13 @@
 @AllArgsConstructor
 @Getter
 public class TeleportData {
-    private final Vector3d location;
-    private final Vector3d velocity;
-    private final RelativeFlag flags;
+    private final double locationX;
+    private final double locationY;
+    private final double locationZ;
+    private final double velocityX;
+    private final double velocityY;
+    private final double velocityZ;
+    private @NotNull final RelativeFlag flags;
```

Comment:
you could also make this a byte if it's a problem
```suggestion
    private final @NotNull RelativeFlag flags;
```